applications/challengedac.md

```diff
@@ -0,0 +1,109 @@
+
+### Ecosystem Fit
+
+ChallengeDapp is a decentralized application (dApp) that runs on the EOS blockchain and it is designed to allow users to create and participate in a variety of challenges. ChallengeDapp utilizes the EOS blockchain to provide a decentralized and trustless platform where users can create and participate in Challenges. The use of smart contracts on the EOS blockchain ensures that the rules of each challenge are transparent and immutable and that the rewards for winning a challenge are automatically distributed to the winners. ChallengeDapp also utilizes the EOS token on the platform. Users can use EOS tokens to create new challenges, participate in existing challenges, and receive rewards for winning challenges. This dApp fits into the EOS ecosystem by providing a platform that can be used to create and participate in various challenges, and by incentivizing users to engage with the EOS blockchain and use EOS tokens. Additionally, it offers entertainment and educational opportunities for users and it can be seen as a way to increase the adoption and usage of the EOS blockchain.
+
+## Team
+
+### Team members
+
+- **Team Leader:** Chace Eskam
+- **Co-founder:** Kent Eskam
+- **System Engineer:** Phung Ta
+
+### Legal Structure
+
+- **Registered Legal Entity:** CALI-PUMP LLC
+- **Registered Address:** 300 Carlsbad Village Dr, Carlsbad, CA 92008
+
+### Team Experience
+
+
+Chace Eskam: Has spent a number of years studying the social and economic impact of the blockchain. Chace has given talks about blockchains future implications on business at W.P Carey school of business as early as February 2016. Before devoting himself fully to blockchain Chace spent a number of years working in the fintech industry at the Nations first digital bank.
+
+### Team Org Repos
+
+- https://github.com/ChallengeDAC
+- https://github.com/ChallengeDAC/dacdart
+
+## Development Roadmap
+
+### Milestone Summary
+
+- **Total Estimated Duration:** 6 months 
+- **Full-Time Equivalent (FTE):** 1 FTE
+- **Total Costs:** 18,500 USD
+
+### Milestone 1 - Develop DacDart, a decentralized SDK, using EOS Node API to hook to ChallengeDac application blocks to view accounts, performs smart contracts
+
+- **Estimated duration:** 2 months
+- **FTE:**  1
+- **Costs:** 9,000 USD
+
+| ID  | Deliverable | Specification |
+| --- | ----------- | ------------- |
+| 1.  | Refactor EOS Node APIs | $7,000 is added to the development of refactoring the EOS Node APIs to seperate modules written in Dart |
```

Comment:
Can you indicate which APIs you need to refactor and specify what you need to change?

Reply:
In the ChallengeDapp, we integrated [eosdart_ecc](https://pub.dev/documentation/eosdart/latest/eosdart/EOSClient-class.html) to hash the json, then sign the hashed data with encryptedPrivateKey and password then it will be sent to our server to perform [smart contracts](https://bloks.io/account/challengeyou?loadContract=true&tab=Actions&account=challengeyou&scope=challengeyou&limit=100). That module is still in our private source code of ChallengeDapp. Our future target is to wrap this into an Dart open-source plugin for (Mobile, Web, Desktop) which aims to helps other developers to integrate ChallengeDapp smart contracts into their aapplications

Reply:
Thank you, but can you please also answer API changes question from @linh2931?

Reply:
Hi. Firstly, we only use [get\_table\_rows](https://developers.eos.io/manuals/eos/v2.2/nodeos/plugins/chain_api_plugin/api-reference/index#operation/get_table_rows) API. Secondly, I apologize for the lack of information. I just change the "Deliverable" name to match our target from the beginning. The module will have a lot of things to do with ECC for EOS private key management rather than EOS Node APIs. I updated the Milestones with more accurate information. Please check it, thank you!